Makefile

```diff
@@ -423,7 +425,7 @@ envoy-bin/envoy-static: $(ENVOY_BASH.deps) FORCE | envoy-bin
 
 check-envoy: $(ENVOY_BASH.deps)
 	$(call ENVOY_BASH.cmd, \
-	    docker exec --workdir=/root/envoy $$(cat envoy-build-container.txt) bazel test --verbose_failures -c dbg --test_env=ENVOY_IP_TEST_VERSIONS=v4only //test/...; \
+	    docker exec $$(cat envoy-build-container.txt) /bin/sh -c 'cd /root/envoy && bazel test --verbose_failures -c dbg --test_env=ENVOY_IP_TEST_VERSIONS=v4only //test/...'; \
```

Comment:
Surprisingly, `--workdir` fails with certain versions of docker even with compatible APIs, hence this change.